json-schema/web_assets.json

```diff
@@ -0,0 +1,61 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+
+  "definitions": {
+    "assets": {
+      "type": "object",
+      "properties": {
+        "name": { "type": "string" },
+        "dependencies": {
+          "type": "array",
+          "contains": {
+            "type": "string"
+          }
+        },
+        "js": {
+          "type": "array",
+          "contains": {
+            "type": "string"
+          }
+        },
+        "css": {
+          "type": "array",
+          "contains": {
+            "type": "string"
+          }
+        },
+        "attribute": {
+          "type": "array",
```

Comment:
`attribute` is an object, the thing like next:
```
"assets": [
  {
  "name": "foo",
  "js": [
      "path/to/foo.js",
      "path/to/foo2.js"
    ],
    "css": [
      "path/to/foo.css",
      "path/to/foo2.css"
    ],
    "attribute":{
      "path/to/foo.js": {
        "type": "module"
      },
      "path/to/foo2.js": {
        "defer": true,
        "data-foo": "bar"
      },
      "path/to/foo.css": {
        "media": "print"
      },
      "path/to/foo2.css": {
        "media": "screen and (max-width: 600px)"
      }
    }
  }
]
```

Reply:
Hmm maybe we need to rethink this structure then :/ because these dynamic keys of file names are really bad for building a schema

Reply:
that was easiest way to code an assigning an attribute per file :wink: 
not sure what we can do differently

Reply:
Maybe just leave it without defining a properties, they said:
> By default any additional properties are allowed.

Reply:
Out of curiosity:
This is fine and it's the current way to deliver ES2015+ scripts
```js

      "path/to/foo.js": {
        "type": "module"
      },
```
Is it possible to also do:
```js
"path/to/foo2.js": {
        "defer": true,
        "nomodule": ""
      },
```
which is the way to have the ES5 code of the same script? (eg imports are flattened here)

Reply:
Supported anything that `$doc->addScript()` supports

Reply:
In final it will be in PHP equivalent:
```php
$doc->addScript('path/to/foo2.js', [...], [
  "defer" => true,
  "nomodule" => ""
])
```

Reply:
OK made it an object. Not convinced it's the best way. But it's a way :)

Reply:
looks good to me :)